src/eth/rpc.ts

```diff
@@ -30,6 +36,20 @@ const getDelegatorSnapshot = async (
     delegate,
   };
 };
+const isEOA = async (address: string) => {
+    return provider.getCode(address) === "0x"
```

Comment:
Looks like [provider.getCode](https://docs.ethers.io/v5/single-page/#/v5/api/providers/provider/-%23-Provider-getCode) returns a Promise so I suspect this line would throw an error since it compares a Promise against a string.

Also, I'm guessing that the original goal was to pass an array of Promises to `Promise.all` in `filterAddresses()`. At the moment, this function doesn't return a Promise so I don't think the `Promise.all` would've worked to begin with. 

May be able to do something like this instead:

```javascript
const isEOA = async (address: string) => {
    return (await provider.getCode(address)) === "0x"
}

const isSnapshotEligible = async (item) => {
    return new Promise<boolean>((resolve) => {
        resolve(isEOA(item.id) && !isOrchestrator(item))
    })
}
```

I recommend testing that out though.

Reply:
> it compares a Promise against a string

my bad 🤦🏻‍♂️

fe175d9 breaks down filtering into 2 parts 

I felt that removing Os is rather simple and does not depend on external calls so might as well do it first and then use that result to filter EOAs. Less complicated and better readability. Conditional checking of a promise and non-promise in the same statement was causing issues.

Reply:
> it compares a Promise against a string

my bad 🤦🏻‍♂️

aefca83 breaks down filtering into 2 parts 

I felt that removing Os is rather simple and does not depend on external calls so might as well do it first and then use that result to filter EOAs. Less complicated and better readability. Conditional checking of a promise and non-promise in the same statement like above was causing issues. WDYT?

Reply:
Breaking up filtering into 2 steps seems fine, but...

See my other comment about Promise.all() not actually receiving an array of Promises right now. Does this actually work?